golem-worker-executor-base/src/durable_host/durability.rs

```diff
@@ -18,62 +18,262 @@ use crate::metrics::wasm::record_host_function_call;
 use crate::model::PersistenceLevel;
 use crate::services::oplog::{CommitLevel, OplogOps};
 use crate::workerctx::WorkerCtx;
+use async_trait::async_trait;
 use bincode::{Decode, Encode};
 use bytes::Bytes;
-use golem_common::model::oplog::{OplogEntry, OplogIndex, WrappedFunctionType};
-use golem_common::serialization::try_deserialize;
+use golem_common::model::oplog::{DurableFunctionType, OplogEntry, OplogIndex};
+use golem_common::model::Timestamp;
+use golem_common::serialization::{serialize, try_deserialize};
+use golem_wasm_rpc::{IntoValue, IntoValueAndType, ValueAndType};
 use std::fmt::Debug;
 use std::marker::PhantomData;
 use tracing::error;
 
+#[derive(Debug)]
+pub struct DurableExecutionState {
+    pub is_live: bool,
+    pub persistence_level: PersistenceLevel,
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct PersistedDurableFunctionInvocation {
+    timestamp: Timestamp,
+    function_name: String,
+    response: Vec<u8>,
```

Comment:
It might make sense to not load the response by default but instead allow streaming them in when needed. I believe we can have quite easily have a situation where response_size > host_memory, especially with streaming responses.

Reply:
It is not in this level - when persisting a streamed wasi-http response (which I think you have in mind) each chunk read is persisted as a separate oplog entry.
This type is only used in host functions made durable, so basically what they do with it is that they read from the oplog what they otherwise would return when not in replay mode.

It is of course interesting to think of what happens when WASI (and WIT etc) itself will support streams - but to persist functions returning streams will require many changes / improvements anyway.